test/x86/sse4.2.c

```diff
@@ -24,9 +24,231 @@
 #define SIMDE_TESTS_CURRENT_ISAX sse4_2
 #include <test/x86/test-x86-internal.h>
 #include <simde/x86/sse4.2.h>
+#include <assert.h>
 
 #if defined(SIMDE_X86_SSE4_2_NATIVE) || defined(SIMDE_NO_NATIVE) || defined(SIMDE_ALWAYS_BUILD_NATIVE_TESTS)
 
+#if defined(SIMDE_X86_SSE4_2_NATIVE)
+#define SIMDE_SIDD_UBYTE_OPS _SIDD_UBYTE_OPS
+#define SIMDE_SIDD_UWORD_OPS _SIDD_UWORD_OPS
+#define SIMDE_SIDD_SBYTE_OPS _SIDD_SBYTE_OPS
+#define SIMDE__SIDD_SWORD_OPS _SIDD_SWORD_OPS
+#define SIMDE_SIDD_CMP_EQUAL_ANY _SIDD_CMP_EQUAL_ANY
+#define SIMDE_SIDD_CMP_RANGES _SIDD_CMP_RANGES
+#define SIMDE_SIDD_CMP_EQUAL_EACH _SIDD_CMP_EQUAL_EACH
+#define SIMDE_SIDD_CMP_EQUAL_ORDERED _SIDD_CMP_EQUAL_ORDERED
+#define SIMDE_SIDD_POSITIVE_POLARITY _SIDD_POSITIVE_POLARITY
+#define SIMDE_SIDD_NEGATIVE_POLARITY _SIDD_NEGATIVE_POLARITY
+#define SIMDE_SIDD_MASKED_POSITIVE_POLARITY _SIDD_MASKED_POSITIVE_POLARITY
+#define SIMDE_SIDD_MASKED_NEGATIVE_POLARITY _SIDD_MASKED_NEGATIVE_POLARITY
+#define SIMDE_SIDD_LEAST_SIGNIFICANT _SIDD_LEAST_SIGNIFICANT
+#define SIMDE_SIDD_MOST_SIGNIFICANT _SIDD_MOST_SIGNIFICANT
+#define SIMDE_SIDD_BIT_MASK _SIDD_BIT_MASK
+#define SIMDE_SIDD_UNIT_MASK _SIDD_UNIT_MASK
+
+#else
+#define SIMDE_SIDD_UBYTE_OPS 0x00
+#define SIMDE_SIDD_UWORD_OPS 0x01
+#define SIMDE_SIDD_SBYTE_OPS 0x02
+#define SIMDE_SIDD_SWORD_OPS 0x03
+#define SIMDE_SIDD_CMP_EQUAL_ANY 0x00
+#define SIMDE_SIDD_CMP_RANGES 0x04
+#define SIMDE_SIDD_CMP_EQUAL_EACH 0x08
+#define SIMDE_SIDD_CMP_EQUAL_ORDERED 0x0c
+#define SIMDE_SIDD_POSITIVE_POLARITY 0x00
+#define SIMDE_SIDD_NEGATIVE_POLARITY 0x10
+#define SIMDE_SIDD_MASKED_POSITIVE_POLARITY 0x20
+#define SIMDE_SIDD_MASKED_NEGATIVE_POLARITY 0x30
+#define SIMDE_SIDD_LEAST_SIGNIFICANT 0x00
+#define SIMDE_SIDD_MOST_SIGNIFICANT 0x40
+#define SIMDE_SIDD_BIT_MASK 0x00
+#define SIMDE_SIDD_UNIT_MASK 0x40
+#endif
```

Comment:
These generally look good, but  they should be in sse4.2.h not the tests.  They're part of the public SSE4.2 API.

They also still need native aliases, otherwise it will fail if the target doesn't support SSE4.2 but `SIMDE_ENABLE_NATIVE_ALIASES` is defined.  You just need to add another set like

```c
#if defined(SIMDE_X86_SSE4_2_ENABLE_NATIVE_ALIASES)
  #define _SIDD_UBYTE_OPS SIMDE_SIDD_UBYTE_OPS
  // ...
#endif
```